casetokens.php

```diff
@@ -144,6 +144,46 @@ function _casetokens_get_case_id() {
   return isset(\Civi::$statics['casetokens']['case_id']) ? \Civi::$statics['casetokens']['case_id'] : NULL;
 }
 
+/**
+ * Get all the contact entity fields.
+ *
+ * @return array
+ */
+function _casetokens_get_contact_fields() {
+  $contactId = CRM_Core_Session::singleton()->getLoggedInContactID();
+  $contactFields = array();
+  try {
+    $contactFields = civicrm_api3('contact', 'getsingle', array(
+      'id' => $contactId,
+    ));
+  } catch (Throwable $ex) {
+  }
+
+  return array_keys($contactFields);
+}
+
+/**
+ * Get all the contact custom fields.
+ *
+ * @return array
+ */
+function _casetokens_get_contact_custom_fields() {
+  try {
+    $customFields = civicrm_api3('CustomField', 'get', array(
+      'custom_group_id.extends' => array('IN' => array("Contact", "Individual", "Household", "Organization")),
+    ));
+  } catch (Throwable $ex) {
+  }
+  $fields = array();
+  if (!empty($customFields) && !empty($customFields['values'])) {
+    foreach ($customFields['values'] as $id => $allFields) {
+      $fields['custom_' . $id] = $allFields['name'];
+    }
+  }
+
+  return $fields;
+}
```

Comment:
I don't think this funcion is necessary. `civicrm_api3('contact', 'getfields')` *will* return custom fields as well as core.

Reply:
does getfields returns all custom fields that extends Contacts, Individual, Organization and Household?

Reply:
Yes

Reply:
@colemanw I have changed the things as per your suggestions. Kindly do let me know if there is anything else that you would like me to change.

Reply:
@colemanw it will really help us if you confirm that you're not happy with this PR and that its good to be approved. We have a deadline by end of day today for this. If you can please approve it by end of today that will be a huge help! :)

Thank you.

Reply:
@sohaib-compucorp I've merged it

Reply:
Awesome, thanks A LOT @colemanw :) you're a star!!!

Reply:
@colemanw if you can do me one last favour please, if you can please release a new version and tag a new release to allow us to be able to use this extension.

Since we have a deadline for this by end of today, if you can do this before end of today it will be a massive help!! :)

Thank you once again for your help with this. Much appriciated. :)